sources/commands/Hydrate.ts

```diff
@@ -36,6 +37,9 @@ export class HydrateCommand extends Command<Context> {
 
     const {default: tar} = await import(/* webpackMode: 'eager' */ `tar`);
 
+    if (!existsSync(installFolder))
+      mkdirSync(installFolder, {recursive: true});
```

Comment:
I don't think we gain anything from the `existSync` check, it's already built-in `mkdir` when `recursive` is `true`.
```suggestion
  // Recreate the folder in case it was deleted somewhere else:
  await mkdir(installFolder, {recursive: true});
```

This needs to be moved below, in the loop, in case the directory is deleted while `corepack` is running, we want to re-create it just before the `tar.x` command runs.